liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -153,28 +152,28 @@ plugin = GHC.defaultPlugin {
 --------------------------------------------------------------------------------
 --
 -- Since we have no good way of passing extra data between different
--- phases of our plugin, we resort to horrible global mutable state.
+-- phases of our plugin, we resort to leaving breadcrumbs in horrible
+-- global mutable state.
 --
 -- Each module gets a mutable breadcrumb during compilation.
 --
 -- The @parseResultAction@ sets the breadcumb to @Parsed mod@.
 --
--- The @typeCheckResultAction@ sets the breadcrumb to @Typechecking@ on entry,
--- and clears the breadcrumb on exit.
+-- The @typeCheckResultAction@ clears the breadcrumb on entry.
 --
 -- If the @typeCheckResultAction@ does not find the breadcrumb, it skips the
 -- module, assuming that it has been verified already. This could happen if
 -- the plugin is activated more than once on the same module (by passing
 -- multiple times @-fplugin=LiquidHaskell@ to GHC).
+{- HLINT ignore Breadcrumb "Use newtype instead of data" -}
```

Comment:
Probably a comment is deserved here to explain why `data` is preferred to `newtype` in this case.